src/utils/flow.js

```diff
@@ -31,7 +31,7 @@ export function assertExhaustiveCheck(
  * type information of the object. Flow will occasionally throw errors when
  * inferring what is going on with Object.assign.
  */
-export function immutableUpdate<T: Object>(object: T, ...rest: Object[]): T {
+export function immutableUpdate<T>(object: T, ...rest: any[]): T {
   return Object.assign({}, object, ...rest);
 }
```

Comment:
I did that originally, but it broke the types. Since it's a bit out of scope of the patch, I'm going to skip, since it might take a bit of investment of time to figure it out.